worker/raft/worker.go

```diff
@@ -232,6 +256,26 @@ func (w *Worker) Raft() (*raft.Raft, error) {
 	return nil, ErrWorkerStopped
 }
 
+// LogStore returns the raft.LogStore managed by this worker, or
+// an error if the worker has stopped.
+func (w *Worker) LogStore() (raft.LogStore, error) {
+	// Give precedence to the worker dying.
```

Comment:
This can just be done like so can't it?

	select { 
	case <-w.catacomb.Dying(): 
                return nil, ErrWorkerStopped
	case logStore := <-w.logStoreCh: 
		return logStore, nil 
	} 

And the stuff following the select can be simplified also

Reply:
I kind of cargo-culted this from the `Raft()` getter above, but I think the rationale is that if both catacomb.Dying() and w.logStoreCh are ready, it's indeterminate which case will be executed. This structure gives priority to the dying case and avoids us handing out a log store or raft object that will probably be useless (closed or shutdown) by the time the client gets to use it. Maybe it makes it more likely that you get one kind of error rather than many different kinds in this situation? I'm not sure it's *that* important though - obviously you could still get the log store and have it get closed under you anyway. I'm ok to simplify it if you like.

Reply:
The case statements are evaluated in order. So case <-Dying() being first will ensure that it gets priority.

Reply:
I'm pretty sure they're not evaluated in order in a select... 
I checked and the language spec agrees with me.
https://golang.org/ref/spec#Select_statements

None of which is to say I shouldn't just simplify it anyway, if we don't need the priority.